superset/utils/machine_auth.py

```diff
@@ -33,10 +33,18 @@
 if TYPE_CHECKING:
     from flask_appbuilder.security.sqla.models import User
 
+    try:
+        from playwright.sync_api import BrowserContext
+    except ModuleNotFoundError:
+        BrowserContext = Any
+
 
 class MachineAuthProvider:
     def __init__(
-        self, auth_webdriver_func_override: Callable[[WebDriver, User], WebDriver]
+        self,
+        auth_webdriver_func_override: Callable[
+            [WebDriver | BrowserContext, User], WebDriver | BrowserContext
+        ],
```

Comment:
Good point. I decided to make separate auth overrides instead of merging - since it's an experimental feature, it will be  easier to quickly enable/disable the FF if the auth functions are separate